### PR TITLE
Feat(eos_designs): RFC5549 support for core_interfaces

### PR DESCRIPTION
**Pull request by emilarista**
## Change Summary

This PR implements RFC5549 underlay capability to core_interfaces data model. Additionally adds a toggle to make ipv4 optional on core_interfaces.

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

No new data model added, changed core_interfaces logic slightly to allow absence of ip/ip_pool/subnet.

## How to test
Tested with molecule and CH-Project repo.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/doc/core-interfaces-BETA.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ core_interfaces:
       # Specific IP addresses used on this P2P link | Optional (Requires ip_pool or subnet or ip)
       ip: [ < node_a IPv4_address/Mask >, < node_b IPv4_address/Mask > ]
 
+      # Allows turning off ipv4 addressing for the link or profile | Optional
+      ipv4_enable: < true | false | default -> true >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/core-interfaces-BETA.md]: wouldn't it be more intuitive to just _not_ define `ip` above? So allow no `ip`, `ip_pool` or `subnet`

Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/doc/core-interfaces-BETA.md]: Very true, I modified the logic to instead just allow not defining any of ip/ip_pool/subnet.

**Review by onurgashi — APPROVED** [review]
LGTM!

File: ansible_collections/arista/avd/roles/eos_designs/templates/core_interfaces/logic.j2 [hunk 2]
```diff
@@ -24,20 +25,20 @@
...
 {%                     break %}
 {%                 endif %}
 {%             endfor %}
 {%         endif %}
 {%         set subnet = p2p_link.subnet | arista.avd.default(
-                        link_info.p2p_profile.subnet) %}
+                            link_info.p2p_profile.subnet, none) %}
 {%         set ip_list = p2p_link.ip | arista.avd.default(
-                         link_info.p2p_profile.ip) %}
+                            link_info.p2p_profile.ip, none) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/templates/core_interfaces/logic.j2]: none/null/None is already the default value returned if the default var is not defined.

File: ansible_collections/arista/avd/molecule/evpn_underlay_rfc5549_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml [hunk 3]
```diff
@@ -183,3 +183,21 @@ bfd_multihop:
   multiplier: 3
 
 custom_structured_configuration_prefix: [ 'my_dci_', 'my_special_dci_', 'custom_structured_configuration_' ]
+
+core_interfaces:
+  p2p_links_profiles:
+    - name: default_bb_profile
+      ipv6_enable: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/evpn_underlay_rfc5549_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml]: IMO this should be autodetected based on `include_in_underlay_protocol` and the `underlay_rfc5549`. It is fine to keep the knob if you want to enable v6 in other cases, but for the 5549 it should just work without it.

File: ansible_collections/arista/avd/roles/eos_designs/templates/core_interfaces/logic.j2 (hunk 2 above)
Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_designs/templates/core_interfaces/logic.j2]: Cool, i removed it again.

File: ansible_collections/arista/avd/molecule/evpn_underlay_rfc5549_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml (hunk 3 above)
Comment (emilarista) [reply ansible_collections/arista/avd/molecule/evpn_underlay_rfc5549_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml]: Made it autodetect based on rfc5549 and include_in_underlay_protocol. Also updated readme to reflect this behavior.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**